Add a param to indicate the redirect URI when back from Google authorization

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

Implemented partial #486 and based on #832.

Please refer to the **Google account reconnection flow** section in the #486 description. For the reconnection step 3, this PR added a param `next` to the **GET** `/wc/gla/google/connect`, thus we can indicate the redirect URI when back from Google authorization.

### Detailed test instructions:

#### The MC onboarding flow should work as before:

1. Request to **GET** `/wc/gla/google/connect` and proceed with the Google authorization through responded `url`
2. After finish the auth, it should redirect the browser back to the MC onboarding `/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsetup-mc&google-mc=connected`

#### The Google reconnection flow:

1. With the new `next` param, request to **GET** `/wc/gla/google/connect?next=reconnect`, and proceed with the Google authorization through responded `url`
2. After finish the auth, it should redirect the browser back to the reconnection page `/wp-admin/admin.php?page=wc-admin&path=/google/settings&subpath=/reconnect-accounts&google-mc=connected`.
3. Please note that it will show the accounts disconnection UI in this PR, and the reconnection page will be added by the following PRs.

### Changelog entry

> Add  - Added a param to indicate the redirect URI when back from Google authorization

eason9487 committed:
Add a param to indicate the redirect URI when back from Google authorization

mikkamp commented on src/API/Site/Controllers/Google/AccountController.php:
```diff
@@ -83,17 +85,38 @@ public function register_routes(): void {
+					'url' => $this->connection->connect( admin_url( "admin.php?page=wc-admin&path=$path" ) ),
 				];
 			} catch ( Exception $e ) {
 				return new WP_REST_Response( [ 'message' => $e->getMessage() ], $e->getCode() ?: 400 );
```
Not related to this PR, but this should be just Response.

```suggestion
				return new Response( [ 'message' => $e->getMessage() ], $e->getCode() ?: 400 );
```

mikkamp commented on src/API/Site/Controllers/Google/AccountController.php:
```diff
@@ -83,17 +85,38 @@ public function register_routes(): void {
+
 				return [
-					'url' => $this->connection->connect( admin_url( 'admin.php?page=wc-admin&path=/google/setup-mc' ) ),
+					'url' => $this->connection->connect( admin_url( "admin.php?page=wc-admin&path=$path" ) ),
```
Small nitpick, which isn't a problem since it's at the end of the string. I think it's better to wrap the variables in curly brackets.

```suggestion
					'url' => $this->connection->connect( admin_url( "admin.php?page=wc-admin&path={$path}" ) ),
```

eason9487 replied:
Thanks for the suggestion.

mikkamp reviewed (approved):
Looks good, I'm redirected to the right URL.
`admin.php?page=wc-admin&path=%2Fgoogle%2Fsettings&subpath=%2Freconnect-accounts&google-mc=connected`

Just added a couple small suggestions.

eason9487 committed:
Apply suggestions from code review

Address
- https://github.com/woocommerce/google-listings-and-ads/pull/833#discussion_r659648381
- https://github.com/woocommerce/google-listings-and-ads/pull/833#discussion_r659651120

Co-authored-by: Mik <mikkamp@users.noreply.github.com>
